crates/cranelift/src/wasmfx/optimized.rs

```diff
@@ -341,19 +341,19 @@ pub(crate) mod typed_continuation_helpers {
 
         pub fn args(&self) -> Payloads {
             let offset = wasmtime_continuations::offsets::continuation_object::ARGS;
-            Payloads::new(self.address, offset, self.pointer_type)
+            Payloads::new(self.address, offset as i32, self.pointer_type)
         }
 
         pub fn tag_return_values(&self) -> Payloads {
             let offset = wasmtime_continuations::offsets::continuation_object::TAG_RETURN_VALUES;
-            Payloads::new(self.address, offset, self.pointer_type)
+            Payloads::new(self.address, offset as i32, self.pointer_type)
```

Comment:
In general `offset as i32` could truncate the offset, since `offset: usize` and `usize` has machine-dependent width. I guess here we are save, because the value of `offset` should never much larger than ~32.

Reply:
indeed